repository/src/main/java/org/pentaho/platform/repository2/unified/webservices/IUnifiedRepositoryWebService.java

```diff
@@ -67,6 +67,10 @@ RepositoryFileDto createFolderWithAcl( final String parentFolderId, final Reposi
   @Deprecated
   List<RepositoryFileDto> getChildren( final String folderId );
 
+  default void doesFileExist( String path ) throws Exception {
```

Comment:
If this function `void doesFileExist( String path ) throws Exception {` is necessary.

And you want some feedback you should never throw the generic class Exception. You should throw a more specific checked exception either from standard java library, an existing Pentaho one or create a new one that serves you purpose.

 boolean canUnlockFile( final String fileId );

Reply:
this is also noted in sonarqube  relating to generic exception